drivers/sensor/vl53l0x/vl53l0x.c

```diff
@@ -140,6 +196,18 @@ static int vl53l0x_setup_single_shot(const struct device *dev)
 	return ret;
 }
 
+static int vl53l0x_set_ranging_mode(const struct device *dev, VL53L0X_DeviceModes mode)
+{
+	struct vl53l0x_data *drv_data = dev->data;
+	int ret = VL53L0X_SetDeviceMode(&drv_data->vl53l0x, mode);
+	if (ret) {
+		LOG_ERR("[%s] VL53L0X_SetDeviceMode failed", dev->name);
+		return ret;
+	}
+	drv_data->current_mode = mode;
+	return 0;
+}
```

Comment:
This change doesn't look related to interrupt-gpios. Please extract to a separate commit.

Reply:
In my opinion, this change is related to interrupt gpios. Before this commit, the sensor was always in single-shot mode (ie the i2c master tells the sensor to start a measurement, waits, then fetch the result from the sensor).

In this commit, we add support for `sensor_trigger_set` for `SENSOR_TRIG_DATA_READY` in the vl53l0x driver. For this trigger to make sense, we need to put the sensor in continuous mode (ie the sensor continuously makes new measurements, and notifies the i2c master via an interrupt that a new measurement is ready).

In continuous mode, the caller does not need to start a new measurement in `sensor_sample_fetch`, so we need to track the current mode of the device to know if a new measurement should be started. This function simply wraps the call to the ST HAL, so that the current device mode is properly tracked. 

Finally, the device mode is not exposed to the user, so this function is only an accompanying helper for the interrupt gpio functionality